### DISCUSSION TITLE
The Current State of the Project

### DISCUSSION THREAD

**Discussion by LeNitrous** [vignetteapp/vignette#104] [Announcements]
Hello there! Nitrous here. I'm one of the lead developers of this project here to give you all a status update regarding the current state of the project.

As you may have noticed in the past few months, we haven't made any new working commit to this repository. Thus, no new working artifacts or builds have been published as of late. First of all in behalf of the team, I apologize for this matter. However I do like to make known some things to everyone looking forward to this project to several things.

1. **The codebase originally was a mess**
    - This originally started as a prototype under the name *holotrack*. We had a working build in under a month at the cost of quality. Thus many bugs and issues arose from this making it unpresentable. It was a real struggle to work with it and since it was hastily put up, it really was unacceptable and proved difficult to deal with it in the long run.

2. **Our libaries needed an overhaul**
    - To put this into perspective, the camera library was hastily put up and the Live2D library used had code dating back to 2018 under .NET Framework 4.5 which we had to hastily port to .NET Core 3. Both libraries proved it didn't provide us much flexibility we needed. This follows up to our next point.

3. **Introducing new features to our libraries for us and other consumers**
    - We already decided upon fully refactoring the codebase on this project. But it wouldn't be fun if we just directly port it to a better state. We have to add new things along the way. For context, it was difficult to test the project without a physical camera. While it can be solved by using one to begin with, our budget didn't allow us for it.
    - In this, we introduced a virtual camera allowing us to use pre-recorded video for face tracking. Other than that, the Live2D library we were using proved difficult to work with osu!framework, the game framework we'd been using for this project. Not to mention it was missing important features such as Inverted Mask support (new in Cubism 4) and physics.

***tl;dr*** We basically had to refactor the entire codebase. Whilst doing so, we added new features to our libraries and future-proofed it for the long-term. I did mention *our libraries* very often. In fact, you too can use it in your own osu!framework projects. While its still in a state of refactoring, you can visit it in the meantime to see if its ready.
- [Camera Support for osu!framework](https://github.com/vignette-project/vignette-application-camera)
- [Live2D Support for osu!framework](https://github.com/vignette-project/vignette-application-live2d)

Thank you for your continued support and hope to do so. We'll check back with everyone here once we know we're ready to release a proper build.

**Comment by sr229** [comment]
Additional Anecdote: We have a tentative release date of 2021, so I opened a [goal](https://ko-fi.com/capuccino) in my Ko-fi to cover notarization, infrastructure and finally the CDNs we require. Additionally, donating regularly on our LiberaPay also does the same effect as well!
